pkg/dependencies/dependents.go

```diff
@@ -0,0 +1,194 @@
+//
+// Copyright 2024 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dependencies
+
+import (
+	"context"
+	"fmt"
+	"github.com/Khan/genqlient/graphql"
+	"github.com/guacsec/guac/cmd/guacone/cmd"
+	model "github.com/guacsec/guac/pkg/assembler/clients/generated"
+	"github.com/guacsec/guac/pkg/handler/collector/deps_dev"
+	"sort"
+)
+
+type dependencyNode struct {
+	dependents map[string]bool // map of the Name of the dependent to whether it is a dependent of the current node
+}
+
+type PackageName struct {
+	Name           string
+	DependentCount int
+}
+
+type Dependencies interface {
+	GetSortedDependents() ([]PackageName, error)
+}
+
+type dependencies struct {
+	ctx       context.Context
+	gqlClient graphql.Client
+}
+
+func New(ctx context.Context, gqlClient graphql.Client) Dependencies {
+	return &dependencies{
+		gqlClient: gqlClient,
+		ctx:       ctx,
+	}
+}
+
+func (deps *dependencies) GetSortedDependents() ([]PackageName, error) {
```

Comment:
The interface and object are all not needed, there is no need to save the state in an object. Just make this a function:
`func GetSortedDependents(ctx context.Context, gqlClient graphql.Client) ([]PackageName, error)`

Reply:
Removed the interface and refactored the function to take in `ctx` and `gqlClient` as parameters.